sdk/helper/ocsp/client.go

```diff
@@ -0,0 +1,1085 @@
+		NextUpdate: time.Unix(int64(cacheValue.nextUpdate), 0).UTC(),
+		Status:     int(cacheValue.status),
+	})
+}
+
+/*
+// writeOCSPCache writes a OCSP Response cache
+func (c *Client) writeOCSPCache(ctx context.Context, storage logical.Storage) error {
+	c.Logger().Debug("writing OCSP Response cache")
+	t := time.Now()
+	m := make(map[string][]interface{})
+	keys := c.ocspResponseCache.Keys()
+	if len(keys) > persistedCacheSize {
+		keys = keys[:persistedCacheSize]
+	}
+	for _, k := range keys {
+		e, ok := c.ocspResponseCache.Get(k)
+		if ok {
+			entry := e.(*ocspCachedResponse)
+			// Don't store if expired
+			if isInValidityRange(t, time.Unix(int64(entry.thisUpdate), 0), time.Unix(int64(entry.nextUpdate), 0)) {
+				key := k.(certIDKey)
+				cacheKeyInBase64, err := decodeCertIDKey(&key)
+				if err != nil {
+					return err
+				}
+				m[cacheKeyInBase64] = []interface{}{entry.status, entry.time, entry.producedAt, entry.thisUpdate, entry.nextUpdate}
+			}
+		}
+	}
+
+	v, err := jsonutil.EncodeJSONAndCompress(m, nil)
+	if err != nil {
+		return err
+	}
+	entry := logical.StorageEntry{
+		Key:   ocspCacheKey,
+		Value: v,
+	}
+	return storage.Put(ctx, &entry)
+}
+
+// readOCSPCache reads a OCSP Response cache from storage
+func (c *Client) readOCSPCache(ctx context.Context, storage logical.Storage) error {
+	c.Logger().Debug("reading OCSP Response cache")
+
+	entry, err := storage.Get(ctx, ocspCacheKey)
+	if err != nil {
+		return err
+	}
+	if entry == nil {
+		return nil
+	}
+	var untypedCache map[string][]interface{}
+
+	err = jsonutil.DecodeJSON(entry.Value, &untypedCache)
+	if err != nil {
+		return errors.New("failed to unmarshal OCSP cache")
+	}
+
+	for k, v := range untypedCache {
+		key, err := c.encodeCertIDKey(k)
+		if err != nil {
+			return err
+		}
+		var times [4]float64
+		for i, t := range v[1:] {
+			if jn, ok := t.(json.Number); ok {
+				times[i], err = jn.Float64()
+				if err != nil {
+					return err
+				}
+			} else {
+				times[i] = t.(float64)
+			}
+		}
+		var status int
+		if jn, ok := v[0].(json.Number); ok {
+			s, err := jn.Int64()
+			if err != nil {
+				return err
+			}
+			status = int(s)
+		} else {
+			status = v[0].(int)
+		}
+
+		c.ocspResponseCache.Add(*key, &ocspCachedResponse{
+			status:     ocspStatusCode(status),
+			time:       times[0],
+			producedAt: times[1],
+			thisUpdate: times[2],
+			nextUpdate: times[3],
+		})
+	}
+
+	return nil
+}
+*/
+
+func New(logFactory func() hclog.Logger, cacheSize int) *Client {
+	if cacheSize < 100 {
+		cacheSize = 100
+	}
+	cache, _ := lru.New2Q(cacheSize)
+	c := Client{
+		caRoot:            make(map[string]*x509.Certificate),
+		ocspResponseCache: cache,
+		logFactory:        logFactory,
+	}
+
+	return &c
+}
+
+func (c *Client) Logger() hclog.Logger {
+	return c.logFactory()
+}
+
+// insecureOcspTransport is the transport object that doesn't do certificate revocation check.
+func newInsecureOcspTransport(extraCas []*x509.Certificate) *http.Transport {
```

Comment:
Should this just go in the test helpers?

Reply:
Now that we have them, yes.

Reply:
Ah, d'oh, sorry. 

I don't like the name then, its not any more insecure than the default Go helper, perhaps just `newTransportWithoutOcsp`?